README.md

```diff
@@ -1,48 +1,68 @@
-# protego
+# Protego
 
-Protego is a tool to permissionlessly deploy a spell which can be used to drop a particular plan in MakerDAO's Pause contract, or in extreme cases the protego contract can itself be lifted to the hat, allowing any user to permissionlessly drop any scheduled plan.
+Protego is a tool to permissionlessly deploy a spell which can be used to drop a particular plan in the `MCD_PAUSE`
+contract, or in extreme cases the `Protego` contract can itself be lifted to the hat, allowing any user to
+permissionlessly drop any scheduled plan.
+
+## Context
+
+- A `plan` is a scheduled `delegatecall` that exists in [`MCD_PAUSE`](https://github.com/makerdao/ds-pause) and can be
+  permissionlessly executed.
+- Plans in `MCD_PAUSE` are [identified by a unique 32 byte hash](https://github.com/makerdao/ds-pause/blob/master/src/pause.sol#L99).
+- A conformant spell here is one where the values returned by `action()`, `tag()`, `sig()` and `eta()` are equal to the
+  corresponding `plan`'s `usr`, `tag`, `fax` and `eta` values respectively. Bad actors could potentially manipulate the
+  return values of these functions such that they are not equal, which would result in a non-conformant spell.
 
 ## Usage
 
-### Drop an individual plan
+### Create a single drop spell
```

Comment:
> I find it necessary to have a very clear instructions (IF/ELSE, step-by-step) explaining which of the 4 functions to use when and in which order.

Fully agree. Part of the deliverable is to provide a long-written post and also train governance facilitators on how to use the tool. We're just not sure the README is the best way to put the content verbatim. Maybe we'll have a link here to these documents once it's finished.

> As the next step, we might even consider writing a script that evaluates currently plotted spells, then fetches required parameters for the specific call and executes it.

Not sure a script would be good enough in this case. There probably needs to be a UI for that and such feature be included there, however this is out of the scope of this repo.

Reply:
My main point is: the choice to use one of the four methods could be included in the scope of the repo specifically for it to be audited. This can be done the simplest way possible, for example, using a `s.sol` file or a js-based script

Reply:
Just to clarify: when you mention the "four methods", you are referring to:

1. Deploy a drop spell with a spell address (for conforming spells)
2. Deploy a drop spell from plot params (for non-conforming spells)
3. Call `Protego.drop()` with a spell address (for conforming spells)
4. Call `Protego.drop()` with plot params (for non-conforming spells)

Correct?

Do you mean we should include all flows into scripts for auditing?
What kind of additional value do you expect this to bring given that those flows are already present in the test files?

Reply:
> Correct?

Correct

> Do you mean we should include all flows into scripts for auditing? What kind of additional value do you expect this to bring, given that those flows are already present in the test files?

Current tests only check listed flows separately, isolated from each other. My proposal is to add usage instruction or (if possible) a script that would explain "how to choose between 4 methods" or directly execute a specific method (e.g. if there are multiple spells plotted and not executed). I propose this because this decision (no matter if described in words or in code) is security-critical (e.g. if a wrong method is used it might not drop the spell) and time-sensitive (highly likely there wouldn't be time to deploy another dropspell as voting also takes time) and therefore would highly benefit from external audit.

Reply:
I'm hoping that the documentation is now clear enough, since there are only 2 methods left.
Maybe there should be something written in a "decision tree" format, but not sure if it's strictly necessary.